### PR TITLE
Issue-426:Convert Pytest-CVP MLAG status check test case to Vane #426

### PR DESCRIPTION
**Pull request by paragpande24**
# Please include a summary of the changes

Describe the solution you'd like
PyTest-CVP MLAG status check test case should have the following:

Use Vane test case styling
Use Master definition file to simplify input
Add test case steps to the test case
Test case should be skipped if MLAG is not configured
Test should be added to top-level directory nrfu_tests
Use markers nrfu_test, interfaces
Use Test ID: NRFU2.4

# Any specific logic/part of code you need extra attention on
   From the output of  **show mlag**,  verified that the mlag status is displayed.

# Include the Issue number and link
https://github.com/aristanetworks/vane/issues/426

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU testcases)

# Effort required on reviewers end

- - [ ] Easy
- - [x] Medium
- - [ ] Hard 

# How Has This Been Tested?
Unit testcases:

 - Tested the same testcase The mlag status is displayed.: Passed

[426-PASS-REPORT.docx](https://github.com/aristanetworks/vane/files/12597383/426-PASS-REPORT.docx)

- Tested the mlag status is not displayed (Tested with Hardcode output): Failed

[426-FAIL-REPORT.docx](https://github.com/aristanetworks/vane/files/12597385/426-FAIL-REPORT.docx)

- Tested with The mlag status is displayed. are not found on the device: Skipped

[426-SKIP-REPORT.docx](https://github.com/aristanetworks/vane/files/12597388/426-SKIP-REPORT.docx)

- Tested the same testcase The mlag status is displayed.: Passed with J2 

[426-PASS-REPORT-WITH-J2.docx](https://github.com/aristanetworks/vane/files/12597394/426-PASS-REPORT-WITH-J2.docx)

**HTML_REPORTS**

[426-REPORT-HTML.zip](https://github.com/aristanetworks/vane/files/12597414/426-REPORT-HTML.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/master_def.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+common_data:
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/master_def.yaml:1]: Update it as per placeholder

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2022 Arista Networks, Inc.  All rights reserved.
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:1]:
```suggestion
# Copyright (c) 2023 Arista Networks, Inc.  All rights reserved.

```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import pytest
+from pyeapi.eapilib import EapiError
+from vane.logger import logger
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_tests
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:17]:
```suggestion
@pytest.mark.nrfu_test

```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        self.output = ""
+        tops.actual_output = {"mlag_details": {}}
+        tops.expected_output = {"mlag_details": {}}
+
+        # Forming output message if test result is passed
+        tops.output_msg = "The mlag status is displayed."
+
+        try:
+            """
+            TS: Running 'show mlag' command on DUT and verifying the mlag status.
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:46]:
```suggestion
            TS: Running 'show mlag' command on DUT and verifying that the MLAG is configured on the device.

```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            dut(dict): details related to a particular DUT
+            tests_definitions(dict): test suite and test case parameters
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"mlag_details": {}}
+        tops.expected_output = {"mlag_details": {}}
+
+        # Forming output message if test result is passed
+        tops.output_msg = "The mlag status is displayed."
```
Comment (AbhijeetGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:42]:
```suggestion
        tops.output_msg = "MLAG is configured on the device."

```
Update at all relevant places

File: nrfu_tests/master_def.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+global_dut_filter: # This stanza is optional. If it is not provided by users than all test cases should be run on each dut.
```
Comment (MaheshGSLAB) [inline nrfu_tests/master_def.yaml:1]: lets remove this master def file

File: nrfu_tests/test_definition.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_interfaces_mlag_status_check
+    description: Testcase for verification of mlag interfaces status check.
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_definition.yaml:4]:
```suggestion
    description: Testcase for verification of MLAG functionality.
```

File: nrfu_tests/test_definition.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    test_id: NRFU2.4
+    show_cmd: show mlag
+    expected_output: null # Forming expected output in test file.
+    test_criteria: MLAG should be configured on the device. # Setting test case’s pass / fail criteria for reporting
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_definition.yaml:8]:
```suggestion
    test_criteria: MLAG should be configured as active, connected and consistent on the device . # Setting test case’s pass / fail criteria for reporting
```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+from vane.logger import logger
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_test
+@pytest.mark.interfaces
+class InterfacesMlagStatusCheckTests:
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:19]:
```suggestion
class MlagStatusTests:

```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    """
+    Testcase for verification of mlag interfaces status check.
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:21]: update as per test def comment

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    """
+
+    dut_parameters = tests_tools.parametrize_duts(TEST_SUITE, test_defs, dut_objs)
+    test_duts = dut_parameters["test_interfaces_mlag_status_check"]["duts"]
+    test_ids = dut_parameters["test_interfaces_mlag_status_check"]["ids"]
+
+    @pytest.mark.parametrize("dut", test_duts, ids=test_ids)
+    def test_interfaces_mlag_status_check(self, dut, tests_definitions):
+        """
+        TD: Testcase for verification of mlag interfaces status check.
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:31]: Update as per test def comment

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+            dut(dict): details related to a particular DUT
+            tests_definitions(dict): test suite and test case parameters
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"mlag_details": {}}
+        tops.expected_output = {"mlag_details": {}}
+
+        # Forming output message if test result is passed
+        tops.output_msg = "MLAG is configured on the device."
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:42]: updated as per test def comment

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            assert output, "MLAG details are not found"
+
+            # collecting actual output.
+            if output["state"] != "disabled":
+                tops.actual_output["mlag_details"].update(
+                    {
+                        "state": output.get("state"),
+                        "negStatus": output.get("negStatus"),
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:64]:
```suggestion
                        "negotiation_status": output.get("negStatus"),

```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+                        "configSanity": output.get("configSanity"),
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:65]:
```suggestion
                        "config_sanity": output.get("configSanity"),

```

**Comment by MaheshGSLAB** [comment]
image


Expected MLAG state is 'active', however actual found as 'unactive'.
Expected MLAG negotiation status is 'connected', however actual found as 'disconnected'.
Expected MLAG config sanity is 'consistent', however actual found as 'idel'.

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+            """
+            output = dut["output"][tops.show_cmd]["json"]
+            logger.info(
+                "On device %s, output of %s command is: \n%s\n",
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            assert output, "MLAG details are not found"
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:57]: dont know that this assert is valid or not. Can you please check on multiple device if we can find no mlag details.

File: nrfu_tests/test_definition.yaml [hunk 16]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_interfaces_mlag_status_check
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_definition.yaml:3]:
```suggestion
  - name: test_interfaces_mlag_status
```

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+                        "negotiation_status": "connected",
+                        "config_sanity": "consistent",
+                    }
+                )
+            else:
+                pytest.skip(f"For {tops.dut_name} MLAG is not configured, hence test skipped.")
+
+            # Output message formation in case of testcase fails.
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = ""
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_interfaces_mlag_status_check.py:80]:
```suggestion
                tops.output_msg = "\n"

```

**Review by MaheshGSLAB — CHANGES_REQUESTED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Review by AbhijeetGSLAB — APPROVED** [review]

**Review by rewati-arista — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+        # Forming output message if test result is passed
+        tops.output_msg = "MLAG is configured as active, connected and consistent on the device."
+
+        try:
+            """
+            TS: Running 'show mlag' command on DUT and verifying that the MLAG is configured
+            on the device.
+            """
+            output = dut["output"][tops.show_cmd]["json"]
+            logger.info(
```
[the remark this answers is not in the document]
Comment (paragpande24) [reply nrfu_tests/test_interfaces_mlag_status_check.py]: Addressed comment.

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+                    if mlag_value != tops.actual_output["mlag_details"].get(mlag_key):
+                        tops.output_msg += (
+                            f"Expected MLAG {mlag_key.replace('_', ' ')} is '{mlag_value}',"
+                            " however actual found as"
+                            f" '{tops.actual_output['mlag_details'].get(mlag_key)}'.\n"
+                        )
+
+        except (AssertionError, AttributeError, LookupError, EapiError) as excep:
+            tops.output_msg = tops.actual_output = str(excep).split("\n", maxsplit=1)[0]
+            logger.error(
```
[the remark this answers is not in the document]
Comment (paragpande24) [reply nrfu_tests/test_interfaces_mlag_status_check.py]: Addressed comment.

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+            on the device.
+            """
+            output = dut["output"][tops.show_cmd]["json"]
+            logger.info(
+                "On device %s, output of %s command is: \n%s\n",
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
```
[the remark this answers is not in the document]
Comment (paragpande24) [reply nrfu_tests/test_interfaces_mlag_status_check.py:53]: As discussed with Kevin, not updating this change, Once the vane issue is fixed for generating a report we can remove this self. output.

File: nrfu_tests/test_interfaces_mlag_status_check.py [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+
+        # Forming output message if test result is passed
+        tops.output_msg = "MLAG is configured as active, connected and consistent on the device."
+
+        try:
+            """
+            TS: Running 'show mlag' command on DUT and verifying that the MLAG is configured
+            on the device.
+            """
+            output = dut["output"][tops.show_cmd]["json"]
```
[the remark this answers is not in the document]
Comment (paragpande24) [reply nrfu_tests/test_interfaces_mlag_status_check.py:49]: As discussed in the call with Kevin and Shachi, they suggested keeping this change as it is.

**Pull request merged**